aggregator/src/aggregator/taskprov_tests.rs

```diff
@@ -78,238 +78,276 @@ pub struct TaskprovTestCase {
+            transcript_2.leader_prepare_transitions[0].message.clone(),
+        )]),
     );
-    let aggregate_resp: AggregationJobResp = decode_response_body(&mut test_conn).await;
+    let aggregation_job_id_2: AggregationJobId = random();
+
+    for (name, request, aggregation_job_id, report_share) in [
+        ("request_1", request_1, aggregation_job_id_1, report_share_1),
+        ("request_2", request_2, aggregation_job_id_2, report_share_2),
+    ] {
+        let auth = test
+            .peer_aggregator
+            .primary_aggregator_auth_token()
+            .request_authentication();
+
+        let mut test_conn = put(test
+            .task
+            .aggregation_job_uri(&aggregation_job_id)
+            .unwrap()
+            .path())
+        .with_request_header(auth.0, "Bearer invalid_token")
+        .with_request_header(
+            KnownHeaderName::ContentType,
+            AggregationJobInitializeReq::<FixedSize>::MEDIA_TYPE,
+        )
+        .with_request_header(
+            TASKPROV_HEADER,
+            URL_SAFE_NO_PAD.encode(test.task_config.get_encoded()),
+        )
+        .with_request_body(request.get_encoded())
+        .run_async(&test.handler)
+        .await;
+        assert_eq!(test_conn.status(), Some(Status::BadRequest), "{}", name);
+        assert_eq!(
+            take_problem_details(&mut test_conn).await,
+            json!({
+                "status": Status::BadRequest as u16,
+                "type": "urn:ietf:params:ppm:dap:error:unauthorizedRequest",
+                "title": "The request's authorization is not valid.",
+                "taskid": format!("{}", test.task_id),
+            }),
+            "{}",
+            name
+        );
+
+        let mut test_conn = put(test
+            .task
+            .aggregation_job_uri(&aggregation_job_id)
+            .unwrap()
+            .path())
+        .with_request_header(auth.0, auth.1)
+        .with_request_header(
+            KnownHeaderName::ContentType,
+            AggregationJobInitializeReq::<FixedSize>::MEDIA_TYPE,
+        )
+        .with_request_header(
+            TASKPROV_HEADER,
+            URL_SAFE_NO_PAD.encode(test.task_config.get_encoded()),
+        )
+        .with_request_body(request.get_encoded())
+        .run_async(&test.handler)
+        .await;
 
-    assert_eq!(aggregate_resp.prepare_resps().len(), 1);
-    let prepare_step = aggregate_resp.prepare_resps().get(0).unwrap();
-    assert_eq!(prepare_step.report_id(), test.report_share.metadata().id());
-    assert_matches!(prepare_step.result(), &PrepareStepResult::Continue { .. });
+        assert_eq!(test_conn.status(), Some(Status::Ok), "{}", name);
+        assert_headers!(
+            &test_conn,
+            "content-type" => (AggregationJobResp::MEDIA_TYPE)
+        );
+        let aggregate_resp: AggregationJobResp = decode_response_body(&mut test_conn).await;
+
+        assert_eq!(aggregate_resp.prepare_resps().len(), 1, "{}", name);
```

Comment:
nit:
```suggestion
        assert_eq!(aggregate_resp.prepare_resps().len(), 1, "{name}");
```
and in a few other places